### PR TITLE
🪟🎨 Connector builder: Show callout when stream listing limit is reached

### LINKED ISSUE #21792: Connector builder: Show warning callout if page/slice limit is reached
Based on https://github.com/airbytehq/airbyte/issues/20771

If the `test_read_limit_reached` flag is set on the test read response, a warning callout should be shown to the user informing them that a "real" read during a sync job might yield more data.

As this callout might be shown often, it should be dismissable

## Acceptance criteria

* Show a warning callout in the testing panel if the test read limit is reached
* Make it dismisable (cross icon in the top right, if clicked it will store a flag in local storage and not show it again)
* Additionally have a little indicator with a tooltip next to the page selector

### PR DESCRIPTION
**Pull request by flash1293**
## What

Fixes #21792 by showing a warning message and a in the testing panel if the stream testing limit is reached (warning can be dismissed and won't come back):

Screenshot 2023-02-08 at 12 22 25
Screenshot 2023-02-08 at 12 22 21

## How
Very similar to how the info box in the test input modal works: https://github.com/airbytehq/airbyte/blob/0cef7b0e55bbfba9ef094f59c914b91549c360ac/airbyte-webapp/src/components/connectorBuilder/StreamTestingPanel/ConfigMenu.tsx#L97

### CODE REVIEW

File: airbyte-webapp/src/locales/en.json [hunk 1]
```diff
@@ -763,6 +763,7 @@
   "connectorBuilder.loadingStreamList": "Loading",
   "connectorBuilder.noStreamSelected": "No stream selected",
   "connectorBuilder.streamListUrlError": "Could not load URL",
+  "connectorBuilder.streamTestLimitReached": "Stream testing limit reached. During testing only at max 5 stream slices with 5 pages each will be returned",
```
Comment (timroes) [inline airbyte-webapp/src/locales/en.json:766]: Nitpick: Could we end the 2nd sentence also with a `.`? It feels a bit weird for me having correct punctuation in the beginning of the tooltip, but not in the end.

**Review by timroes — APPROVED** [review]
Code LGTM. Have not tested locally.

**Comment by cgardens** [comment]
@flash1293 the build isn't passing on this PR. can we get a green build before saying it is #ready-to-ship , please?

**Comment by flash1293** [comment]
@cgardens sorry for that, CO ran into rate limits. I updated with master again, if they go green I will add the label again for tomorrow.

**Comment by cgardens** [comment]
Thanks!

**Pull request merged**